app/public/js/localSprites.js

```diff
@@ -141,6 +144,35 @@ function updateLocalSprite(localSprite) {
         // This flag will be used in the update loop to remove this sprite from the list of localSprites.
         localSprite.shouldBeRemoved = true;
     }
+    //geomtry collision checks for projectiles that die on impact
+    //BROKEN: It looks to me like "alignToTile" isn't happening, and the fireball is detonating in the frame before it would impact something,
+    //  rather than right next to the wall it would be moving past.
+    if (localSprite.vx && localSprite.diesOnImpact) {
+        if (localSprite.vx < 0) {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vx, TILE_LEFT))) localSprite.shouldBeRemoved = true;
+        } else {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vx, TILE_RIGHT))) localSprite.shouldBeRemoved = true;
+        }
+    }
+    if (localSprite.vy && localSprite.diesOnImpact) {
+        if (localSprite.vy < 0) {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vy, TILE_UP))) localSprite.shouldBeRemoved = true;
+        } else {
+            if (!(moveSpriteInDirection(localSprite, localSprite.vy, TILE_DOWN))) localSprite.shouldBeRemoved = true;
+        }
+    }
+    //update trigger zones
+    if (localSprite.name === 'triggerZone') {
+        if (rectanglesOverlap(localSprite.hitBox, getGlobalSpriteHitBox(localSprite.target)) && localSprite.cooldownTimer === 0) { //WRONG, probably: sould probabaly use getGlobalSpriteHitBox(localSprite), but not sure how that will interact with triggerZones lack of meaningful frames data.
+            if (localSprite.spawnedObjectType === 0) addHomingFireballSprite(localSprite.spawnedObjectXOffset, localSprite.spawnedObjectYOffset, localSprite.target);
```

Comment:
This should probably use a constants, like I do for TILE_BOUNCE or TILE_DAMAGE. So in one part of the code you would do:
```
OBJECT_TYPE_FIREBALL = 0;
OBJECT_TYPE_HEART = 1;
```
etc, then here you would check
`if (localSprite.spawnedObjectType === OBJECT_TYPE_FIREBALL)`